packages/expo-prevent-screen-capture/android/src/main/kotlin/expo/modules/preventscreencapture/PreventScreenCaptureModule.kt

```diff
@@ -0,0 +1,68 @@
+package expo.modules.preventscreencapture
+
+import android.app.Activity
+import android.content.Context
+import android.view.WindowManager
+
+import org.unimodules.core.ExportedModule
+import org.unimodules.core.ModuleRegistry
+import org.unimodules.core.errors.CurrentActivityNotFoundException
+import org.unimodules.core.Promise
+import org.unimodules.core.interfaces.ExpoMethod
+import org.unimodules.core.interfaces.ActivityProvider
+
+class PreventScreenCaptureModule(context: Context) : ExportedModule(context) {
+
+  private lateinit var activityProvider: ActivityProvider
+
+  override fun getName(): String {
+    return NAME
+  }
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    activityProvider = moduleRegistry.getModule(ActivityProvider::class.java)
+  }
+
+  @ExpoMethod
+  fun activatePreventScreenCapture(promise: Promise) {
+    val activity = getCurrentActivity()
+
+    activity.runOnUiThread{
+      try {
+        activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_SECURE)
+        promise.resolve(null)
+      } catch(exception: Exception) {
+        promise.reject(ERROR_TAG, "Failed to prevent screen capture." + exception)
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun deactivatePreventScreenCapture(promise: Promise) {
+    val activity = getCurrentActivity()
+    
+    activity.runOnUiThread{
+      try {
+        activity.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SECURE)
+        promise.resolve(null)
+      } catch(exception: Exception) {
+        promise.reject(ERROR_TAG, "Failed to reallow screen capture." + exception)
+      }
+    }
+  }
+
+  @Throws(IllegalStateException::class)
+  fun getCurrentActivity(): Activity {
+    val activity = activityProvider.currentActivity
+    if (activity != null) {
+      return activity
+    } else {
+      throw IllegalStateException("No activity found.")
```

Comment:
`@unimodules/core` even has a special exception for it! https://github.com/expo/expo/blob/94cbc77e990f54bf92219750616209c085714ff5/packages/@unimodules/core/android/src/main/java/org/unimodules/core/errors/CurrentActivityNotFoundException.java#L5